stock_location_orderpoint/models/stock_move.py

```diff
@@ -9,3 +11,66 @@ class StockMove(models.Model):
     location_orderpoint_id = fields.Many2one(
         "stock.location.orderpoint", "Stock location orderpoint"
     )
+
+    def _prepare_location_orderpoint_replenishment(self, location_field, domain):
+        locations_products = {}
+        for move in self:
+            if not move.filtered_domain(domain):
+                continue
+            location = getattr(move, location_field)
+            locations_products.setdefault(location, self.env["product.product"])
+            locations_products[location] |= move.product_id
+
+        for location, products in locations_products.items():
+            for product in products:
+                self._enqueue_auto_replenishment(
+                    location, product, location_field
+                ).delay()
+
+    def _enqueue_auto_replenishment(
+        self, location, product, location_field, **job_options
+    ):
+        """Enqueue a job stock.location.orderpoint.moves_auto_replenishment()
+
+        Can be extended to pass different options to the job (priority, ...).
+        The usage of `.setdefault` allows to override the options set by default.
+
+        return: a `Job` instance
+        """
+        job_options = job_options.copy()
+        job_options.setdefault(
+            "description",
+            _("Try to replenish quantities for location {} and product {}").format(
+                location.display_name, product.display_name
+            ),
+        )
+        # do not enqueue 2 jobs for the same location and product set
+        job_options.setdefault("identity_key", identity_exact)
+        delayable = self.env["stock.location.orderpoint"].delayable(**job_options)
+        job = delayable.run_auto_replenishment(
+            location,
+            product,
+            location_field == "location_id" and location_field or "location_src_id",
+        )
+        return job
+
+    def _action_confirm(self, *args, **kwargs):
+        moves = super()._action_confirm(*args, **kwargs)
+        moves._prepare_location_orderpoint_replenishment(
+            "location_id",
+            self.env["stock.location.orderpoint"]._get_waiting_move_domain(),
+        )
+        return moves
+
+    def _action_done(self, *args, **kwargs):
+        moves = super()._action_done(*args, **kwargs)
+        moves._prepare_location_orderpoint_replenishment(
```

Comment:
Can you add a comment with the intent?
Why not having the domain in a method like the other case?
This is to trigger when the stock increases on the location serving for the replenishment?
Shouldn't be on the move lines? Otherwise you need to include view location
I need to test, not clear to me

Reply:
If a move is done which increases the stock on a source location of a configured orderpoints
it runs a replenishment for it